utils/ami-cleanup/ami_cleanup.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+import boto3
+
+# Snapshot description filters
+snap_filter_1="Created by CreateImage*for AMI_ID*"
+snap_filter_2="Copied for DestinationAmi AMI_ID*"
+
+# Initialise logging
+logger = logging.getLogger(__name__)
+log_level = os.environ["LOG_LEVEL"] if "LOG_LEVEL" in os.environ else "INFO"
+logger.setLevel(logging.getLevelName(log_level.upper()))
+logging.basicConfig(
+    stream=sys.stdout,
+    format="%(asctime)s %(levelname)s %(module)s "
+           "%(process)s[%(thread)s] %(message)s",
+)
+logger.info("Logging at {} level".format(log_level.upper()))
+
+def get_arguments():
+    parser = argparse.ArgumentParser(description='Delete older AMIs except for N latest.')
+
+    parser.add_argument('--aws-profile', help='AWS profile ID for the account containing AMIs', required=True)
+    parser.add_argument('--aws-region', help='AWS region', required=True)
+    parser.add_argument('--ami-prefix', help='AMI name prefix', required=True)
+    parser.add_argument('--keep-min', type=int, help='Minimum number of AMIs to keep', required=True)
+    parser.add_argument('--aws-profile-list', nargs='*', help='List of additional AWS profile IDs to check AMI usage in')
```

Comment:
This confused me a little to begin with, may change the description?
`List of additional AWS profile IDs to check AMI usage in` ->
`List of additional AWS profile IDs to use to check for AMI usage in other accounts, i.e. accounts with which the AMI is shared`